needy/caches/directory.py

```diff
@@ -0,0 +1,92 @@
+import os
+import shutil
+import time
+import fcntl
+
+from ..cache import Cache, KeyLocked, SourceNotFound, KeyNotFound
+from ..filesystem import clean_file, lock_file
+
+
+class Directory(Cache):
```

Comment:
it looks like safety is up to the user with `lock_key` / `unlock_key`. be sure to document that. probably at the `Cache` class